### PR TITLE
JENKINS-44926 prevent stackoverflow in TryBlueOceanMenu

### PR DESCRIPTION
**Pull request by i386**
# Description

See [JENKINS-44926](https://issues.jenkins-ci.org/browse/JENKINS-44926).

# Submitter checklist
- [x] Link to JIRA ticket in description, if appropriate.
- [x] Change is code complete and matches issue description
- [x] Appropriate unit or acceptance tests or explanation to why this change has no tests
- [x] Reviewer's manual test instructions provided in PR description. See Reviewer's first task below.

# Reviewer checklist
- [ ] Run the changes and verified the change matches the issue description
- [ ] Reviewed the code
- [ ] Verified that the appropriate tests have been written or valid explanation given

### CODE REVIEW

**Comment by i386** [comment]
@vivek I have send the reporter a patched HPI and this seems to fix the stack overflow issue. Can you please review and merge to 1.1.2?

File: blueocean-rest-impl/src/main/java/io/jenkins/blueocean/service/embedded/TryBlueOceanMenu.java [hunk 1]
```diff
@@ -60,14 +60,14 @@
...
-        for(Action a: actionable.getActions()) {
-            String blueUrl  = blueOceanUrlAction.getUrlName();
-            String thisUrl = a.getUrlName();
-            if(a instanceof BlueOceanUrlAction
-                    && thisUrl != null
-                    && blueUrl != null
-                    && thisUrl.equals(blueUrl)){
-                return true;
+        for(Action a: actionable.getAllActions()) {
+            // JENKINS-44926 only call getURLName on these actions if action is BLueOceanUrlAction
```
Comment (vivek) [inline blueocean-rest-impl/src/main/java/io/jenkins/blueocean/service/embedded/TryBlueOceanMenu.java]: Ouch, yeah calling any method on a random action is dangerous. Thanks for the fix!

Comment (vivek) [inline blueocean-rest-impl/src/main/java/io/jenkins/blueocean/service/embedded/TryBlueOceanMenu.java]: Ouch, yeah calling any method on a random action is dangerous. Thanks for the fix!

**Comment by i386** [comment]
@vivek thanks!

**Comment by i386** [comment]
This has been cherry picked to master

**Pull request merged**